conda/activate.py

```diff
@@ -618,6 +618,38 @@ def _get_starting_path_list(self):
     def _get_path_dirs(self, prefix):
         if on_win:  # pragma: unix no cover
             yield prefix.rstrip("\\")
+
+            # We need to stat(2) for possible environments because
+            # tests can't be told where to look!
+            #
+            # mingw-w64 is a legacy variant used by m2w64-* packages
+            #
+            # We could include clang32 and mingw32 variants
+            variants = []
+            for variant in ["ucrt64", "clang64", "mingw64", "clangarm64"]:
+                path = self.sep.join((prefix, "Library", variant))
+
+                # MSYS2 /c/
+                # cygwin /cygdrive/c/
+                if re.match("^(/[A-Za-z]/|/cygdrive/[A-Za-z]/).*", prefix):
+                    path = unix_path_to_native(path)
+
+                if isdir(path):
+                    variants.append(variant)
+
+            if len(variants) > 1:
+                print(
+                    f"WARNING: {prefix}: {variants} MSYS2 envs exist: please check your dependencies",
```

Comment:
@ifitchet Would it be better for end-users to see an example of wrong dependencies in the environment? When I saw this warning message I couldn't get what it means

Reply:
@skupr-anaconda probably.

However, I don't know how to extract such information.  What I do have is the list of possible variants in my hands so at least I can give _some_ clues as to why they won't necessarily see all the packages they have installed on their PATH.

If, though, that is confusing, then it is less useful.